Replicator/Replicator.cc

```diff
@@ -1307,4 +1295,40 @@ namespace litecore { namespace repl {
         }
     }
 
+    // This method is to properly initialize the passive replicator to get ready to
+    // serve 3.0 replicator, which is unaware of the collection.
+    // It is a no-op if
+    // 1. it is working in the active mode,                             or
+    // 2. the incoming meesage includes explicit "collection" property, or
+    // 3. the second time and after that this method is called.
+    void Replicator::setMsgHandlerFor3_0_Client(Retained<blip::MessageIn> request)
+    {
+        if (_setMsgHandlerFor3_0_ClientDone) {
+            return;
+        } else {
+            _setMsgHandlerFor3_0_ClientDone = true;
+        }
+
+        if (_options->isActive()) {
+            return; // only deal with passive replicator.
+        }
+
+        if (request->intProperty(kCollectionProperty, kNotCollectionIndex)
+            != kNotCollectionIndex) {
+            return; // 3.0 message should not include the collection property
+        }
+
```

Comment:
```suggestion
// At this point, we are dealing with a 3.0 style replicator which can only have exactly one collection, which is the default one.  If the default collection is not specified in the passive config (or the default collection does not exist), then we return an error.
```

Reply:
Sure sure how this "commit suggestion" work, and, since I need to do some other changes, plus little modification of the above comment, I copy-pasted the comment to my local machine and pushed a new revision.